core/src/main/resources/META-INF/patches/groovy/02-fix-providerKey-constraints.resolved.groovy

```diff
@@ -0,0 +1,422 @@
+         def resultSet = statement.executeQuery()) {
+
+        while (resultSet.next()) {
+            def id = resultSet.getString("id")
+            def providerKey = resultSet.getString("providerKey")
+            duplicates[providerKey] << id
+        }
+    } catch (SQLException e) {
+        e.printStackTrace()
+    }
+
+    return duplicates
+}
+
+/**
+ * Test duplicate insert, we will try to insert two times the same provider key
+ * In order to check if database is correctly configured or not
+ * @param dataSource
+ * @return true if duplicate insert is allowed, false otherwise
+ */
+static def checkIfDuplicatedInsertIsPossible(DataSource dataSource) {
+    def providerKey = UUID.randomUUID().toString()
+    def insertQuery = "INSERT INTO jahia_external_provider_id (providerKey) VALUES (?)"
+    def deleteQuery = "DELETE FROM jahia_external_provider_id WHERE providerKey = ?"
+
+    dataSource.connection.withCloseable { conn ->
+        conn.autoCommit = false
+        try {
+            conn.prepareStatement(insertQuery).withCloseable { stmt ->
+                stmt.setString(1, providerKey)
+                stmt.executeUpdate()
+                stmt.setString(1, providerKey)
+                stmt.executeUpdate()
+            }
+            conn.commit()
+            return true
+        } catch (SQLException ignored) {
+            conn.rollback()
+            return false
+        } finally {
+            conn.prepareStatement(deleteQuery).withCloseable { stmt ->
+                stmt.setString(1, providerKey)
+                stmt.executeUpdate()
+            }
+        }
+    }
+}
+
+/**
+ * Get the count of external uuids where internalUuid is using given provider id.
+ * @param dataSource The database connection source.
+ * @param providerId The ID of the provider.
+ * @return The count of matching external uuids.
+ */
+static def countMatchingInternalUuid(DataSource dataSource, String providerId) {
+    def paddedId = StringUtils.leftPad(providerId, 8, "f")
+    def query = """
+        SELECT COUNT(*) AS count 
+        FROM jahia_external_mapping 
+        WHERE internalUuid LIKE ?
+    """
+
+    def count = 0
+    try (def connection = dataSource.connection;
+         def statement = connection.prepareStatement(query)) {
+        statement.setString(1, "${paddedId}%")
+        try (def resultSet = statement.executeQuery()) {
+            if (resultSet.next()) {
+                count = resultSet.getInt("count")
+            }
+        }
+    }
+
+    return count
+}
+
+/**
+ * Deletes all records from the jahia_external_provider_id table where the primary key (id) matches the provided id.
+ * @param dataSource The database connection source.
+ * @param providerId The primary key value to match for deletion.
+ */
+static def deleteProviderById(DataSource dataSource, int providerId, boolean deleteExternalMappings = false) {
+    def queryProvider = "DELETE FROM jahia_external_provider_id WHERE id = ?"
```

Comment:
find better name for varialbe, as this is not just a query, but a delete statment